src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeColorization.cs

```diff
@@ -111,54 +128,67 @@ internal static void UpdateApplicationResources(Uri dictionaryUri)
+            }
         }
 
         _currentApplicationTheme = themeToApply;
         _currentAppsTheme = GetAppsTheme();
     }
 
-    internal static void ApplyTheme(Windows.Window window)
+    private static Uri GetApplicationThemeUri(string systemTheme, out ApplicationTheme applicationTheme)
     {
-        string themeToApply = GetSystemTheme();
+        string themeFileName = "light.xaml";
 
-        if (IsThemeDark() && Utility.IsOSWindows11OrNewer)
-        {
-            UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "dark.xaml", UriKind.Absolute));
-            WindowBackgroundManager.UpdateBackground(window, ApplicationTheme.Dark, WindowBackdropType.Mica, false);
+        if(IsThemeDark())
+        {  
+            applicationTheme = ApplicationTheme.Dark;
+            themeFileName = "dark.xaml";
         }
-        else if(SystemParameters.HighContrast && Utility.IsOSWindows11OrNewer)
+        else if(SystemParameters.HighContrast)
         {
-            if(themeToApply.Contains("hcwhite"))
-            {
-                UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "hcwhite.xaml", UriKind.Absolute));
-                WindowBackgroundManager.UpdateBackground(window, ApplicationTheme.HighContrast, WindowBackdropType.None, false);
-            }
-            else if(themeToApply.Contains("hcblack"))
+            applicationTheme = ApplicationTheme.HighContrast;
+            themeFileName = systemTheme switch 
             {
-                UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "hcblack.xaml", UriKind.Absolute));
-                WindowBackgroundManager.UpdateBackground(window, ApplicationTheme.HighContrast, WindowBackdropType.None, false);
-            }
-            else if (themeToApply.Contains("hc1"))
-            {
-                UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "hc1.xaml", UriKind.Absolute));
-                WindowBackgroundManager.UpdateBackground(window, ApplicationTheme.HighContrast, WindowBackdropType.None, false);
-            }
-            else
-            {
-                UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "hc2.xaml", UriKind.Absolute));
-                WindowBackgroundManager.UpdateBackground(window, ApplicationTheme.HighContrast, WindowBackdropType.None, false);
-            }
+                "hcwhite" => "hcwhite.xaml",
+                "hcblack" => "hcblack.xaml",
+                "hc1" => "hc1.xaml",
+                _ => "hc2.xaml"
+            };
         }
-        else if (Utility.IsOSWindows11OrNewer)
```

Comment:
Shouldn't these checks still be present? We can call `ApplyTheme` method from `SystemResources`. The only check we are doing is if fluentwindows is enabled. What if developer tries to use the fluentwindows styles without having required OS support? I am not sure of this case.

Reply:
We have the check in ApplyTheme right ? I don't see the issue.